### PR TITLE
Typos, edited some sentences, removed redundant whitespace

### PR DESCRIPTION
**Pull request by lufriem**

### CODE REVIEW

**Comment by robertosfield** [comment]
Thanks for you efforts of fixing all my typos, I have mild dyslexia make errors and often don't spot them easily, so appreciate helping fixing things.

Unfortunately this PR came in just after I received another typos patch from another contributor.  I have applied this as a branch and have now merged with vsgTutorial master.  Unfortunately github is now flagging up quite a few conflicts which we'll need to review.

I have other tasks to get on with today so will need to come back to this.

**Comment by lufriem** [comment]
> Unfortunately this PR came in just after I received another typos patch from another contributor. I have applied this as a branch and have now merged with vsgTutorial master. Unfortunately github is now flagging up quite a few conflicts which we'll need to review.
> 
> I have other tasks to get on with today so will need to come back to this.

I just resolved all these conflicts (that was a nice crosscheck!), this PR should be updated (if I understand github correctly)

File: 2_Foundations/Serialization.md [hunk 1]
```diff
@@ -84,11 +84,11 @@ void writeValue(const char* propertyName, T value);
...
+When writing out objects you can simply call **object->write(output)** and the appropriate serialization will be invoked, but when you need to serialize back into a file, the appropriate objects have to be created before their **object->read(input)** method can be invoked to read the object members. The way the VulkanSceneGraph provides a means for creating objects on demand is via the [vsg::ObjectFactory](https://github.com/vsg-dev/VulkanSceneGraph/tree/master/include/vsg/io/ObjectFactory.h#L24) singleton class, where only one instance of the Factory exists. The vsg::ObjectFactory is an example of the [Factory Method Design Pattern](https://en.wikipedia.org/wiki/Factory_method_pattern) and [Singleton Design Pattern](https://en.wikipedia.org/wiki/Singleton_pattern).
```
Comment (robertosfield) [inline 2_Foundations/Serialization.md]: The original is correct.  I think you must have scanned it, but not actually read what the paragraph was trying to convey.

File: 2_Foundations/Visitors.md [hunk 2]
```diff
@@ -97,7 +97,7 @@ This cascading simplifies implementations so they only need to override specific
 
 ## Traversal under your control
 
-By design none of the default apply(..) methods provided by vsg::Visitor and vsg::ConstVisitor undertake any type of traversal, the decision on which objects to traverse and how to traverse them is left to visitor subclasses.  The vsg::Object::traverse(..) method can be used by Visitor subclasses to handle traversal of an object's children when this is required, or Visitor subclasses can implement their own traversal of an object's children.
+By design none of the default apply(..) methods provided by vsg::Visitor and vsg::ConstVisitor undertake any type of traversal, the decision on which objects to traverse and how to traverse them is left to visitor subclasses. The vsg::Object::traverse(..) method can be used by Visitor subclasses to handle traversal of an object's children when this is required, or Visitor subclasses can implement their own traversal of an object's children.
```
Comment (robertosfield) [inline 2_Foundations/Visitors.md]: The space after a . is a readability aid, it helps give a sentence a bit more clarity,  As a dyslexic aids like this do help and why I put them in.

File: 2_Foundations/Serialization.md (hunk 1 above)
Comment (lufriem) [reply 2_Foundations/Serialization.md]: Yes you're right.

File: 2_Foundations/Visitors.md (hunk 2 above)
Comment (lufriem) [reply 2_Foundations/Visitors.md]: Ah ok, the documentation is inconsistent in that regard (note that this was changed in lots of other places as well). Maybe some kind of 'documentation style guide' would help here?

Comment (robertosfield) [reply 2_Foundations/Visitors.md]: Well the changes to the VSG docs to remove spaces after . was also wrong but I'm so busy didn't have the energy to argue every single all of the time.

**Pull request merged**